packages/ckeditor5-table/src/tablelayout/tablelayoutui.ts

```diff
@@ -116,5 +128,121 @@ export default class TableLayoutUI extends Plugin {
 
 			return menuView;
 		} );
+
+		// Create table type dropdown button.
+		editor.ui.componentFactory.add( 'tableType', () => {
+			const editor = this.editor;
+			const t = editor.t;
+
+			const button = new DropdownButtonView( editor.locale );
+
+			button.set( {
+				label: t( 'Table type' ),
+				icon: IconTableProperties,
+				tooltip: true
+			} );
+
+			return createTableTypeDropdown( editor, button );
+		} );
 	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public afterInit(): void {
+		const editor = this.editor;
+
+		if ( !editor.plugins.has( 'TablePropertiesUI' ) ) {
+			return;
+		}
+
+		const tablePropertiesUI = editor.plugins.get( 'TablePropertiesUI' );
+
+		// Override the default table properties button to include the table type dropdown.
+		editor.ui.componentFactory.add( 'tableProperties', locale => {
```

Comment:
Please add a code comment explaining why we are registering this component in `afterInit`.